src/test/java/org/springframework/data/couchbase/core/CouchbaseTemplateKeyValueIntegrationTests.java

```diff
@@ -132,6 +184,55 @@ void insertById() {
 
 	}
 
+	@Test
+	void insertByIdwithDurability() {
+		User user = new User(UUID.randomUUID().toString(), "firstname", "lastname");
+		User inserted = couchbaseTemplate.insertById(User.class).withDurability(PersistTo.ACTIVE, ReplicateTo.NONE)
+				.one(user);
+		assertEquals(user, inserted);
+
+		assertThrows(DuplicateKeyException.class, () -> couchbaseTemplate.insertById(User.class).one(user));
+		couchbaseTemplate.removeById().one(user.getId());
+
+	}
+
+	@Test
+	void insertByIdwithExpiry() {
+		User user = new User(UUID.randomUUID().toString(), "firstname", "lastname");
+		try {
+			User inserted = couchbaseTemplate.insertById(User.class).withExpiry(Duration.ofSeconds(1)).one(user);
+			assertEquals(user, inserted);
+			sleepSecs(2);
```

Comment:
are these sleeps needed in those places? kv should not be eventually consistent but rather directly

Reply:
The sleeps are need to allow the document to expire.

Reply:
of course 🤦